### PR TITLE
feat: add `methods` to enable or disable specific null-ls source method types

### PR DESCRIPTION
**Pull request by mehalter**
This potentially supersedes #93 . It takes a slightly different approach to make it easier to modify individual methods. Let me know what you think! I did some performance testing and it doesn't seem to make any significant impact even with a ton of packages getting setup.

This approach also provides nice completion when using the types in the lua language server

### CODE REVIEW

File: lua/mason-null-ls/init.lua [hunk 1]
```diff
@@ -20,6 +20,13 @@ local function setup_handlers(handlers)
 
 	local default_handler = Optional.of_nilable(handlers[1]):or_(_.always(Optional.of_nilable(M.default_setup)))
 
+	local methods = {}
+	for method, enabled in pairs(require('mason-null-ls.settings').current.methods) do
```
Comment (jay-babu) [inline lua/mason-null-ls/init.lua:24]: This looks good. I was thinking of doing the func_or_extend. Where if a function is passed, we give the entire methods that the plugins default to. It also takes a list where we just override to what the user passed in. I think it addresses both the concerns you had too?

Comment (mehalter) [reply lua/mason-null-ls/init.lua:24]: Hm with functions it's tricky to account for them when extending plugin specs like if you have several opts. Plus the function notation in lazy for specifying opts can sorta take care of it. What do you think? I think this new approach of a dictionary makes it easy on all accounts

Comment (mehalter) [reply lua/mason-null-ls/init.lua:24]: Totally up to you though!

Comment (mehalter) [reply lua/mason-null-ls/init.lua:24]: I've been thinking about this a lot and I think just having this dictionary is the simplest solution and is the easiest to extended in an additive or reductive manner. I'm not sure if there is a need for extending it through a function. What do you think?

**Review by jay-babu — APPROVED** [review]

File: lua/mason-null-ls/init.lua (hunk 1 above)
Comment (jay-babu) [reply lua/mason-null-ls/init.lua:24]: Yeah I agree. Sorry just been busy. Merging now

Comment (mehalter) [reply lua/mason-null-ls/init.lua:24]: No problem at all, thanks so much for the consideration ❤️

**Pull request merged**